src/pg_to_evalscript/javascript_datacube/DataCube.js

```diff
@@ -100,6 +108,10 @@ class DataCube {
     }
 
     flattenToArray() {
+        if ((!this.data.shape || this.data.shape.length === 0) && this.data.data.length === 1) {
+            // If there is no data.shape or it's [], we have a scalar.
```

Comment:
There is actually a problem with this - namely this is used on the return from the last node in `evaluatePixel` but `Failed to evaluate script!  evaluatePixel must return an array`. I could swear not long ago it was allowed to return numbers as well (at least if you had one band), although I probably misremember.

However, this is only an issue for returning the data, so I think this is the correct behaviour of `flattenToArray` (because it could be used someone else as well).

So I think we should instead just check in the last line of `evaluatePixel`:
```js
const final = last_node.flattenToArray();
return Array.isArray(final) ? final : [final]
```

Reply:
Was fixed in https://github.com/openEOPlatform/openeo-pg-evalscript-converter/pull/66 as it's not directly connected to the `DataCube`.